### DISCUSSION TITLE
Microsoft Excel ingestion

### DISCUSSION THREAD

**Discussion by BrunoBosshard** [imartinez/privateGPT#1412] [Q&A]
When I try to ingest Microsoft Excel (*.xlsx) files, I get this error message:

**'utf-8' codec can't decode bytes in position 15-16: invalid continuation byte**

The files are in English language, just standard latin characters.

Is there any way around this, maybe a different document loader?

**Comment by hossein890** [comment]
Any answer?

**Comment by mastnacek** [comment]
I have same problem with docx. Maybe some Microsoft garbage? I modified py script which load docs. If end with error, make line into log and continue with other doc.

**Reply by lolo9538** [reply]
Same, I think the try except should be added to the code of the reprository

**Comment by SuperSonnix71** [comment]
The error you're seeing suggests that you are decoding or processing a file with the wrong character encoding. In the context of **ingesting** Microsoft Excel (*.xlsx) files, this error occurs if you’re trying to read the file as if it were a plain text file or using an incorrect method that implicitly assumes a text encoding.

Excel files (.xlsx) are actually a collection of XML files compressed into a single ZIP package. Therefore, trying to read them directly as UTF-8 encoded text files will lead to errors like the one you’re seeing.
Install an use pandas or
openpyxl.

**Reply by TobiasJu** [reply]
Can you elaborate on pandas or openpyxl? 
Do you mean to extract Excel tables into plain text and then ingest them?
